ruby-ci-setup/action.yml

```diff
@@ -0,0 +1,112 @@
+name: "Ruby CI Setup"
+description: "Setup PostgreSQL, LavinMQ, and Ruby environment for CI"
+author: "84codes"
+
+inputs:
+  # Ruby inputs
+  ruby-version:
+    description: "The Ruby version to use"
+    default: ".ruby-version"
+    required: false
+  bundler-cache:
+    description: "Enable bundler caching"
+    default: "true"
+    required: false
+  cache-version:
+    description: "Cache version for cache busting"
+    default: "0"
+    required: false
+  bundle-frozen:
+    description: "Value for BUNDLE_FROZEN environment variable"
+    default: "true"
+    required: false
+
+  # GitHub tokens
+  github-token:
+    description: "GitHub token for private repo dependencies"
+    required: false
+  pkg-github-com:
+    description: "Token for rubygems.pkg.github.com"
+    required: false
+  pkg-github-com-user:
+    description: "GitHub username for rubygems.pkg.github.com auth"
+    default: "machine-user-84"
+    required: false
+
+  # PostgreSQL inputs
+  postgres:
+    description: "Start PostgreSQL in the GitHub Actions VM"
+    default: "false"
+    required: false
+
+  # LavinMQ input
+  lavinmq:
+    description: "Setup LavinMQ"
+    default: "false"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Start PostgreSQL
+      if: inputs.postgres == 'true'
+      shell: bash
+      run: |
+        docker run -d \
+          --name postgres \
+          -e POSTGRES_PASSWORD=password \
+          -e POSTGRES_DB=postgres \
+          -p 5432:5432 \
+          --health-cmd="pg_isready -U postgres" \
+          --health-interval=5s \
+          --health-timeout=5s \
+          --health-retries=5 \
+          postgres:15
+
+    - name: Wait for PostgreSQL to be ready
+      if: inputs.postgres == 'true'
+      shell: bash
+      run: |
+        until docker exec postgres pg_isready -U postgres; do
+          echo "Waiting for PostgreSQL..."
+          sleep 2
+        done
+
+    - name: Setup PostgreSQL extensions
+      if: inputs.postgres == 'true'
+      shell: bash
+      run: |
+        psql -d postgresql://postgres:password@localhost:5432/template1 -c 'CREATE EXTENSION IF NOT EXISTS pgcrypto;'
+        psql -d postgresql://postgres:password@localhost:5432/template1 -c 'CREATE EXTENSION IF NOT EXISTS hstore;'
+        psql -d postgresql://postgres:password@localhost:5432/template1 -c 'CREATE EXTENSION IF NOT EXISTS "uuid-ossp";'
```

Comment:
I wonder if it is good that projects specify this, might be missed for real if not? "Works in CI but not production"?

Reply:
Should probably add it to the intrastructure playbook